### PR TITLE
Generic Shoot Functions

### PR DESCRIPTION
**Pull request by TrevorBrunette**
Shoot action creator functions accept line of sight plus either a handle for the unit or a handle for tank death

### CODE REVIEW

File: src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java [hunk 1]
```diff
@@ -117,42 +121,44 @@ public static final PlayerActionRule<Council> GetRuleCofferCostGrantLife(int cos
...
+                if (optionalUnit.isEmpty()) {
+                    throw new Error(String.format("Target position %s is not on the game board", target.toString()));
+                }
+                IUnit unit = optionalUnit.get();
+
+                if (hit) {
+                    handleHit.accept(s, t, unit);
+                }
+            },
+            new ShootPositionRange("target"),
```
Comment (ryan3r) [inline src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java:142]: Use generic line of site

**Review by ryan3r — CHANGES_REQUESTED** [review]

File: src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java (hunk 1 above)
Comment (TrevorBrunette) [reply src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java]: DONE

File: src/main/java/pro/trevor/tankgame/rule/impl/version3/range/ShootPositionRange.java [hunk 2]
```diff
@@ -6,17 +6,18 @@
...
-        super(name, (state, tank) -> getShootable(state, tank.getPosition(), tank.getRange()));
+    public ShootPositionRange(String name, ITriPredicate<State, Position, Position> lineOfSight) {
+        super(name, (state, tank) -> getShootable(lineOfSight, state, tank.getPosition(), tank.getRange()));
     }
 
-    private static Set<Position> getShootable(State state, Position center, int range) {
+    private static Set<Position> getShootable(ITriPredicate<State, Position, Position> lineOfSight, State state, Position center, int range) {
         Set<Position> output = new HashSet<>();
         for (Position pos : Util.getSpacesInRange(center, range)) {
             if (LineOfSight.hasLineOfSightV3(state, center, pos)) {
```
Comment (ryan3r) [inline src/main/java/pro/trevor/tankgame/rule/impl/version3/range/ShootPositionRange.java:23]: lineOfSight lower cased?

**Review by ryan3r — CHANGES_REQUESTED** [review]

File: src/main/java/pro/trevor/tankgame/rule/impl/version3/range/ShootPositionRange.java (hunk 2 above)
Comment (TrevorBrunette) [reply src/main/java/pro/trevor/tankgame/rule/impl/version3/range/ShootPositionRange.java:23]: Why would it be uppercase?

**Review by ryan3r — APPROVED** [review]

**Pull request merged**